test/image_processing/lanczos_scaling.cpp

```diff
@@ -0,0 +1,82 @@
+//#include <boost/gil/image_processing/scaling.hpp>
+
+#include <boost/gil/extension/io/png.hpp>
+#include <boost/gil/extension/io/jpeg.hpp>
+#include <boost/gil/io/read_image.hpp>
+#include <boost/gil/io/write_view.hpp>
+
+#include <boost/gil/image.hpp>
+#include <boost/gil/image_view.hpp>
+#include <boost/gil/rgb.hpp>
+#include <boost/gil/pixel.hpp>
+
+#include <boost/mpl/int.hpp>
+
+#include <cmath>
+#include <iostream>
+
+namespace gil = boost::gil;
+namespace mpl = boost::mpl;
+using image_view = decltype(gil::view(std::declval<gil::rgb8_image_t&>()));
```

Comment:
This may be enough for a quick prototype of an algorithm, but eventually all of the functionality needs to be as generic as possible, i.e. applicable to as wide a range of channel / pixel / image types as possible.